### ISSUE TITLE
multi_ont_to_json_kg.py fails

### ISSUE THREAD

**Issue by kvarforl** [RTXteam/RTX#1102]
Here is the error:
```
  1 Traceback (most recent call last):
  2   File "/home/ubuntu/kg2-code/multi_ont_to_json_kg.py", line 1204, in <module>
  3     save_pickle)
  4   File "/home/ubuntu/kg2-code/multi_ont_to_json_kg.py", line 147, in make_kg2
  5     curie_to_uri_expander)
  6   File "/home/ubuntu/kg2-code/multi_ont_to_json_kg.py", line 492, in make_nodes_dict_from_ontologies_list
  7     [biolink_category_tree, mappings_to_categories] = generate_biolink_category_tree(first_ontology, curies_to_cat    egories)
  8   File "/home/ubuntu/kg2-code/multi_ont_to_json_kg.py", line 354, in generate_biolink_category_tree
  9     biolink_category_tree = kg2_util.get_biolink_category_tree(biolink_ontology)
 10   File "/home/ubuntu/RTX/code/kg2/kg2_util.py", line 366, in get_biolink_category_tree
 11     parent = biolink_ontology.node(parent)['lbl']
 12   File "/home/ubuntu/kg2-venv/lib/python3.7/site-packages/ontobio/ontol.py", line 341, in node
 13     return self.get_graph().node[id]
 14   File "/home/ubuntu/kg2-venv/lib/python3.7/site-packages/networkx/classes/reportviews.py", line 178, in __getitem    __
 15     return self._nodes[n]
 16 KeyError: 'https://w3id.org/biolink/biolinkml/meta/NamedThing'
```

**Comment by kvarforl** [comment]
Not sure this how relevant this is, but noting that the log for `build-mulit-ont-kg.sh` is filled with these errors (even though they didn't directly kill the script)
![image](https://user-images.githubusercontent.com/39020520/95602257-80fd4f00-0a09-11eb-834c-1894dadaba9e.png)

**Comment by saramsey** [comment]
I am studying build artifacts on `kg2lindsey.rtx.ai`. According to the logfile `kg2-build/build-multi-ont-kg.log`, the script `multi_ont_to_json_kg.py` seems to have terminated shortly after reading `genepio.json`:
```
tail build-multi-ont-kg.log
file: pr.owl; last modified: 2020-10-07 00:16:53 GMT
Reading ontology file: ino.owl; size: 409.75 KiB
/usr/bin/java -Xms2G -Xmx249G -DentityExpansionLimit=4086000 -Djava.awt.headless=true -classpath /home/ubuntu/kg2-build/owltools owltools.cli.CommandLineInterface ino.owl -o -f json /tmp/kg2-0nq363hx.json
Reading ontology JSON file: ino.json; size: 495.96 KiB
file: ino.owl; last modified: 2020-10-07 00:27:04 GMT
Reading ontology file: genepio.owl; size: 894.60 KiB
/usr/bin/java -Xms2G -Xmx249G -DentityExpansionLimit=4086000 -Djava.awt.headless=true -classpath /home/ubuntu/kg2-build/owltools owltools.cli.CommandLineInterface genepio.owl -o -f json /tmp/kg2-87ag67da.json
Reading ontology JSON file: genepio.json; size: 4835.80 KiB
file: genepio.owl; last modified: 2020-10-07 00:27:09 GMT
Calling make_nodes_dict_from_ontologies_list
```

**Comment by saramsey** [comment]
OK, I have constructed a MRE on `kg2lindsey.rtx.ai`, based on a trivial `ont-load-inventory-issue1102.yaml` file that just loads `biolink-model.owl`:
```
- # maps to CURIE prefix: biolink
  url: https://raw.githubusercontent.com/biolink/biolink-model/master/biolink-model.owl
  file: biolink-model.owl
  download: true
  title: Biolink meta-model
```
Running it like this:
```
source ~/kg2-venv/bin/activate
export PATH=$PATH:~/kg2-build
cd ~/kg2-build
python -u ~/kg2-code/multi_ont_to_json_kg.py ~/kg2-code/curies-to-categories.yaml ~/kg2-code/curies-to-urls-map.yaml ~/kg2-code/ont-load-inventory-issue1102.yaml issue1102.json
```
causes this error:
```
Reading ontology file: biolink-model.owl; size: 266.70 KiB
Reading ontology JSON file: biolink-model.json; size: 473.33 KiB
file: biolink-model.owl; last modified: 2020-10-03 17:58:15 GMT
Calling make_nodes_dict_from_ontologies_list
Traceback (most recent call last):
  File "/home/ubuntu/kg2-code/multi_ont_to_json_kg.py", line 1204, in <module>
    save_pickle)
  File "/home/ubuntu/kg2-code/multi_ont_to_json_kg.py", line 147, in make_kg2
    curie_to_uri_expander)
  File "/home/ubuntu/kg2-code/multi_ont_to_json_kg.py", line 492, in make_nodes_dict_from_ontologies_list
    [biolink_category_tree, mappings_to_categories] = generate_biolink_category_tree(first_ontology, curies_to_categories)
  File "/home/ubuntu/kg2-code/multi_ont_to_json_kg.py", line 354, in generate_biolink_category_tree
    biolink_category_tree = kg2_util.get_biolink_category_tree(biolink_ontology)
  File "/home/ubuntu/RTX/code/kg2/kg2_util.py", line 366, in get_biolink_category_tree
    parent = biolink_ontology.node(parent)['lbl']
  File "/home/ubuntu/kg2-venv/lib/python3.7/site-packages/ontobio/ontol.py", line 341, in node
    return self.get_graph().node[id]
  File "/home/ubuntu/kg2-venv/lib/python3.7/site-packages/networkx/classes/reportviews.py", line 178, in __getitem__
    return self._nodes[n]
KeyError: 'https://w3id.org/biolink/biolinkml/meta/NamedThing'
```

**Comment by saramsey** [comment]
Pretty sure that I can reproduce this error on my MBP. Testing now....  yes, this MRE is reproducible on my MBP:
```
sramsey$ python -u multi_ont_to_json_kg.py  curies-to-categories.yaml curies-to-urls-map.yaml  ont-load-inventory-issue1102.yaml issue1102.json
Reading ontology file: biolink-model.owl; size: 266.70 KiB
Reading ontology JSON file: biolink-model.json; size: 473.33 KiB
file: biolink-model.owl; last modified: 2020-10-03 00:26:55 UTC
Calling make_nodes_dict_from_ontologies_list
Traceback (most recent call last):
  File "multi_ont_to_json_kg.py", line 1204, in <module>
    save_pickle)
  File "multi_ont_to_json_kg.py", line 147, in make_kg2
    curie_to_uri_expander)
  File "multi_ont_to_json_kg.py", line 492, in make_nodes_dict_from_ontologies_list
    [biolink_category_tree, mappings_to_categories] = generate_biolink_category_tree(first_ontology, curies_to_categories)
  File "multi_ont_to_json_kg.py", line 354, in generate_biolink_category_tree
    biolink_category_tree = kg2_util.get_biolink_category_tree(biolink_ontology)
  File "/Users/sramsey/Work/Proj/ncats-translator/git-master/RTX/code/kg2/kg2_util.py", line 366, in get_biolink_category_tree
    parent = biolink_ontology.node(parent)['lbl']
  File "/Users/sramsey/Work/Proj/ncats-translator/git-master/RTX/code/kg2/venv/lib/python3.7/site-packages/ontobio/ontol.py", line 341, in node
    return self.get_graph().node[id]
  File "/Users/sramsey/Work/Proj/ncats-translator/git-master/RTX/code/kg2/venv/lib/python3.7/site-packages/networkx/classes/reportviews.py", line 178, in __getitem__
    return self._nodes[n]
KeyError: 'https://w3id.org/biolink/biolinkml/meta/NamedThing'
```
so I am going to shut down `kg2lindsey.rtx.ai` since I can continue to work this issue on the laptop.

**Comment by saramsey** [comment]
Fairly confident this is caused by the same changes to the upstream `biolink-model.owl` file that caused #1085.

**Comment by saramsey** [comment]
Fix 7a269e5 should do it. With that fix, my MRE no longer causes this bug and we are getting a full dictionary of depths for Biolink categories back from `kg2_util.get_biolink_categories_ontology_depths`.

**Comment by saramsey** [comment]
Just confirmed that this fix works on `kg2lindsey.rtx.ai`; the MRE is no longer erroring out.  

@kvarforl you are clear to restart the instance `kg2lindsey.rtx.ai` and resume the build.

**Comment by kvarforl** [comment]
okay we've made it further this time! Again, according to `kg2-build/build-multi-ont-kg.log`, the script `multi_ont_to_json_kg.py`  still seems to have terminated shortly after reading genepio.json.
![image](https://user-images.githubusercontent.com/39020520/95643865-f51a1000-0a66-11eb-953b-9504ecb68be8.png)

However, the error message in `kg2-build/build-kg2-ont-stderr.log` is now the following:
```
2252 Traceback (most recent call last):
2253   File "/home/ubuntu/kg2-code/multi_ont_to_json_kg.py", line 1204, in <module>
2254     save_pickle)
2255   File "/home/ubuntu/kg2-code/multi_ont_to_json_kg.py", line 147, in make_kg2
2256     curie_to_uri_expander)
2257   File "/home/ubuntu/kg2-code/multi_ont_to_json_kg.py", line 657, in make_nodes_dict_from_ontologies_list
2258     bpv_pred_curie = convert_bpv_pred_to_curie_func(bpv_pred)
2259   File "/home/ubuntu/kg2-code/multi_ont_to_json_kg.py", line 1167, in <lambda>
2260     curie_expander)
2261   File "/home/ubuntu/kg2-code/multi_ont_to_json_kg.py", line 1158, in convert_bpv_predicate_to_curie
2262     raise ValueError('unable to expand CURIE: ' + bpv_pred)
2263 ValueError: unable to expand CURIE: MONARCH:cliqueLeader

```

**Comment by saramsey** [comment]
starting `kg2lindsey.rtx.ai` to investigate

**Comment by saramsey** [comment]
OK, this found the problem:
```
cd ~/kg2-build
ubuntu@ip-172-31-59-26:~/kg2-build$ grep -l cliqueLeader *.owl
mondo.owl
```
and then this:
```
ubuntu@ip-172-31-59-26:~/kg2-build$ grep cliqueLeader mondo.owl | head -2
    <!-- https://monarchinitiative.org/MONARCH_cliqueLeader -->
    <owl:AnnotationProperty rdf:about="https://monarchinitiative.org/MONARCH_cliqueLeader"/>
```
so we need to add the following to `curies-to-urls-map.yaml`, at L208:
```
 -
    MONARCH: https://monarchinitiative.org/MONARCH_
```

**Comment by saramsey** [comment]
should be fixed now; just pushed fix bf21817 to `kg2lindsey.rtx.ai`.

**Comment by kvarforl** [comment]
This part of the build process successfully completed 10/13/2020. Closing this issue
